inet/src/inet/common/queue/DropTailQueue.cc

```diff
@@ -109,24 +113,28 @@ cMessage *DropTailQueue::enqueue(cMessage *msg)
-                    logFile << simTime() << " grant packet at droptail enqueue: " << parentHost->getName() << " " << grandParentHost->getName() << " " << grandGrandParentHost->getName() << " " << grandGrandParentHost->getIndex() << std::endl;
-                    logFile.flush();
-                    break;
-                case PktType::SCHED_DATA:
-                    logFile << simTime() << " sched data packet at droptail enqueue: " << parentHost->getName() << " " << grandParentHost->getName() << " " << grandGrandParentHost->getName() << " " << grandGrandParentHost->getIndex() << std::endl;
-                    logFile.flush();
-                    break;
-                case PktType::UNSCHED_DATA:
-                    
-                    break;
-                // default:
-                //     throw cRuntimeError("HomaPkt arrived at the queue has unknown type.");
+            if(homaPkt->getMsgId() == 288){
+                switch (homaPkt->getPktType()) {
+                    case PktType::REQUEST:
+                        logFile2 << simTime() << "len: " << queue.length() << " bytelen: " << queue.getByteLength() << " msg id: " << homaPkt->getMsgId() << " req packet at droptail enqueue: " << parentHost->getName() << " " << grandParentHost->getName() << " " << grandGrandParentHost->getName() << " " << grandGrandParentHost->getIndex() << std::endl;
+                        logFile2.flush();
+                        break;
+                    case PktType::GRANT:
+                        logFile2 << simTime() << "len: " << queue.length() << " bytelen: " << queue.getByteLength() << " msg id: " << homaPkt->getMsgId() << " grant packet at droptail enqueue: " << parentHost->getName() << " " << grandParentHost->getName() << " " << grandGrandParentHost->getName() << " " << grandGrandParentHost->getIndex() << std::endl;
+                        logFile2.flush();
+                        break;
+                    case PktType::SCHED_DATA:
+                        logFile2 << simTime() << "len: " << queue.length() << " bytelen: " << queue.getByteLength() << " msg id: " << homaPkt->getMsgId() << " sched data packet at droptail enqueue: " << parentHost->getName() << " " << grandParentHost->getName() << " " << grandGrandParentHost->getName() << " " << grandGrandParentHost->getIndex() << " firstbyte: " << homaPkt->getSchedDataFields().firstByte << std::endl;
+                        logFile2.flush();
+                        break;
+                    case PktType::UNSCHED_DATA:
+                        logFile2 << simTime() << "len: " << queue.length() << " bytelen: " << queue.getByteLength() << " msg id: " << homaPkt->getMsgId() << " unsched data packet at droptail enqueue: " << parentHost->getName() << " " << grandParentHost->getName() << " " << grandGrandParentHost->getName() << " " << grandGrandParentHost->getIndex() << " firstbyte: " << homaPkt->getUnschedFields().firstByte << std::endl;
+                        logFile2.flush();
```

Comment:
Why do you need to flush? this is single threaded program, logs will eventually find their way to file.